app/models/product_import/entry_validator.rb

```diff
@@ -243,7 +243,7 @@ def mark_as_new_product(entry)
       new_product.assign_attributes(entry.attributes.except('id'))
       new_product.supplier_id = entry.producer_id
 
-      if new_product.valid?
+      if new_product.valid? && new_product.errors.empty?
```

Comment:
I don't understand this extra check. If the model is valid it won't have errors in the array (unless we add them ourselves outside of validations).

Reply:
Ah, I think I added these whilst debugging because the outcome wasn't making sense. I've removed them now. :+1: